compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -86,44 +97,45 @@ generateSrcFromLf env = noLoc mod
         , Just methodName <- [getClassMethodName fieldName]
         ]
 
-    classDecls :: [LHsDecl GhcPs]
+    classDecls :: [Gen (LHsDecl GhcPs)]
     classDecls = do
         LF.DefTypeSyn{..} <- NM.toList . LF.moduleSynonyms $ envMod env
         LF.TStruct fields <- [synType]
         LF.TypeSynName [name] <- [synName]
 
-        let supers =
-                [ convType env fieldType
+        let occName = mkOccName clsName . T.unpack $ sanitize name
+        pure $ do
+            supers <- mapM (convType env)
+                [ fieldType
                 | (fieldName, LF.TUnit LF.:-> fieldType) <- fields
                 , isSuperClassField fieldName
                 ]
-
-            methods =
-                [ (methodName, convType env fieldType)
+            methods <- mapM (secondM (convType env))
+                [ (methodName,fieldType)
```

Comment:
```suggestion
                sequence [ (methodName,) <$> convTyp env fieldType
```

Reply:
that was my first version of this code, hehe :-)